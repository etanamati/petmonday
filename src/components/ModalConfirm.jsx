import React from 'react';
import {Button, Modal} from 'react-bootstrap';

const ModalConfirm = (props) => {
  const {show, title, subTitle, content, handleClose, handleConfirm} = props;
  return (
      <div>
        <Modal show={show} onHide={handleClose}>
          <Modal.Header closeButton>
            <Modal.Title>{title}</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <h4>
              {subTitle}
            </h4>
            <p>
              {content}
            </p>
          </Modal.Body>
          <Modal.Footer>
            <Button variant="secundary" onClick={handleClose}>Fechar</Button>
            <Button variant="success" onClick={handleConfirm}>Confirmar</Button>
          </Modal.Footer>
        </Modal>
      </div>
    )
}

export default ModalConfirm;
