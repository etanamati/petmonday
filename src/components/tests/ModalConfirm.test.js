import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ModalConfirm from '../ModalConfirm';

it('should render app', () => {
  const wrapper = shallow(<ModalConfirm show={true} title='ModalConfirm' subTitle='Sub TÃ­tulo' content='Teste'/>);
    
  expect(toJson(wrapper)).toMatchSnapshot();
});