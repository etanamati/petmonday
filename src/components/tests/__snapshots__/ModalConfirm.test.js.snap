// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app 1`] = `
<div>
  <ForwardRef(Bootstrap(Modal))
    show={true}
  >
    <ForwardRef(Bootstrap(ModalHeader))
      closeButton={true}
    >
      <ForwardRef(Bootstrap(ModalTitle))>
        ModalConfirm
      </ForwardRef(Bootstrap(ModalTitle))>
    </ForwardRef(Bootstrap(ModalHeader))>
    <ForwardRef(Bootstrap(ModalBody))>
      <h4>
        Sub TÃ­tulo
      </h4>
      <p>
        Teste
      </p>
    </ForwardRef(Bootstrap(ModalBody))>
    <ForwardRef(Bootstrap(ModalFooter))>
      <ForwardRef(Bootstrap(Button))
        variant="secundary"
      >
        Fechar
      </ForwardRef(Bootstrap(Button))>
      <ForwardRef(Bootstrap(Button))
        variant="success"
      >
        Confirmar
      </ForwardRef(Bootstrap(Button))>
    </ForwardRef(Bootstrap(ModalFooter))>
  </ForwardRef(Bootstrap(Modal))>
</div>
`;
